extensions/ql-vscode/src/bqrs-cli-types.ts

```diff
@@ -87,15 +75,28 @@ export interface WholeFileLocation {
   startColumn: never;
   endLine: never;
   endColumn: never;
-  charOffset: never;
-  charLength: never;
 }
 
-export type UrlValue = LineColumnLocation | OffsetLengthLocation | WholeFileLocation | string;
+export type UrlValue = LineColumnLocation | WholeFileLocation | string;
 
+export type ResolvableLocationValue = WholeFileLocation | LineColumnLocation;
 
 export type ColumnValue = EntityValue | number | string | boolean;
 
+export type ResultRow = ColumnValue[];
+
+export interface RawResultSet {
+  readonly schema: ResultSetSchema;
+  readonly rows: readonly ResultRow[];
+}
+
+export function adaptBqrs(schema: ResultSetSchema, page: DecodedBqrsChunk): RawResultSet {
```

Comment:
We still want a comment here, do we not, that this function is similarly deprecated to how all of `adapt.ts` was before? I still am of the belief that we could still push further and have code downstream consume `scheme` and `page` more or less directly, and avoid this repackaging. (not that I am asking for that refactoring to be in this PR) 

Alternatively if you think this function should stay, maybe find it a more meaningful name...

Reply:
Yeah. I wasn't happy about keeping this function. It makes the object shape slightly more intuitive, but it is just an extra step that is unnecessary. I'll add a comment here and perhaps we can remove in the future.

Reply:
I'll keep it for now since it makes things slightly easier to work with, though I'm not entirely happy with it. I'll mark it with a TODO that we may want to change in the future.